### PR TITLE
feat(ns-workflows-1): Add Step Object

### PR DESCRIPTION
**Pull request by frankkilcommins**
Refs: #3392

Add Step Object to apidom-ns-workflows-1

### CODE REVIEW

File: packages/apidom-ns-workflows-1/src/refractor/visitors/workflows-1/ParametersVisitor.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import stampit from 'stampit';
```
Comment (char0n) [inline packages/apidom-ns-workflows-1/src/refractor/visitors/workflows-1/ParametersVisitor.ts:1]: Should be located in `workflows-1/step/ParametersVisitor.ts`

File: packages/apidom-ns-workflows-1/src/refractor/visitors/workflows-1/StepOnFailureVisitor.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
+import stampit from 'stampit';
```
Comment (char0n) [inline packages/apidom-ns-workflows-1/src/refractor/visitors/workflows-1/StepOnFailureVisitor.ts:1]: should be located in `workflows-1/step/OnFailureVisitor.ts` and the name of visitor should be `OnFailureVisitor`.

File: packages/apidom-ns-workflows-1/src/refractor/visitors/workflows-1/StepOnSuccessVisitor.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import stampit from 'stampit';
```
Comment (char0n) [inline packages/apidom-ns-workflows-1/src/refractor/visitors/workflows-1/StepOnSuccessVisitor.ts:1]: should be located in `workflows-1/step/OnSuccessVisitor.ts` and the name of visitor should be `OnSuccessVisitor`.

File: packages/apidom-ns-workflows-1/src/refractor/visitors/workflows-1/StepSuccessCriteriaVisitor.ts [hunk 4]
```diff
@@ -0,0 +1,28 @@
+import stampit from 'stampit';
```
Comment (char0n) [inline packages/apidom-ns-workflows-1/src/refractor/visitors/workflows-1/StepSuccessCriteriaVisitor.ts:1]: should be located in `workflows-1/step/SuccessCriteriaVisitor.ts` and the name of visitor should be `SuccessCriteriaVisitor`.

File: packages/apidom-ns-workflows-1/src/refractor/visitors/workflows-1/step/ParametersVisitor.ts [hunk 5]
```diff
@@ -0,0 +1,12 @@
+import stampit from 'stampit';
+
+import StepParametersElement from '../../../../elements/nces/StepParameters';
+import BaseParametersVisitor from '../ParametersVisitor';
+
+const ParametersVisitor = stampit(BaseParametersVisitor, {
```
Comment (char0n) [inline packages/apidom-ns-workflows-1/src/refractor/visitors/workflows-1/step/ParametersVisitor.ts:6]: No need for doing this. There is only one Parameter Object list inside the spec. The BaseParameterVisitor, should be merged into this file: https://github.com/swagger-api/apidom/pull/3471/files#diff-2c8b689e039b2987e9c7a4e8ae33567fa4edcbb89f72cc2f5d614c57927cf873

File: packages/apidom-ns-workflows-1/src/refractor/visitors/workflows-1/ParametersVisitor.ts (hunk 1 above)
Comment (frankkilcommins) [reply packages/apidom-ns-workflows-1/src/refractor/visitors/workflows-1/ParametersVisitor.ts:1]: Thanks - nice spot. Location changed

File: packages/apidom-ns-workflows-1/src/refractor/visitors/workflows-1/StepOnFailureVisitor.ts (hunk 2 above)
Comment (frankkilcommins) [reply packages/apidom-ns-workflows-1/src/refractor/visitors/workflows-1/StepOnFailureVisitor.ts]: Thanks - nice spot. Location changed

File: packages/apidom-ns-workflows-1/src/refractor/visitors/workflows-1/StepSuccessCriteriaVisitor.ts (hunk 4 above)
Comment (frankkilcommins) [reply packages/apidom-ns-workflows-1/src/refractor/visitors/workflows-1/StepSuccessCriteriaVisitor.ts]: Thanks - nice spot.

File: packages/apidom-ns-workflows-1/src/refractor/visitors/workflows-1/step/ParametersVisitor.ts (hunk 5 above)
Comment (frankkilcommins) [reply packages/apidom-ns-workflows-1/src/refractor/visitors/workflows-1/step/ParametersVisitor.ts]: Changes applied - thanks

File: packages/apidom-ns-workflows-1/test/refractor/elements/Step/index.ts [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+      specify('should refract to semantic ApiDOM tree', function () {
+        const stepElement = StepElement.refract({
+          description: 'Search for available pets',
+          stepId: 'searchForPet',
+          operationId: 'getPets',
+          operationRef:
+            'https://petstore3.swagger.io/api/v3/openapi.json#/paths/users/~findbystatus~1{status}/get',
+          workflowId: 'uniqueWorkflowId',
+          parameters: [
+            {
```
Comment (char0n) [inline packages/apidom-ns-workflows-1/test/refractor/elements/Step/index.ts:18]: Missing Reference Object test. I'll follow up with a small change.

**Review by char0n — APPROVED** [review]
Looks good, I'll follow up with one small change.

**Pull request merged**